### ISSUE TITLE
Daemonize option

### ISSUE THREAD

**Issue by jordanisaacs** [ifreund/waylock#49]
It would be nice to have a daemonize option for waylock (similar to swaylock's -f option). It would daemonize itself after finishing setting up the lock.

The reason for this is I would like to use waylock with swayidle. Swayidle allows you to wait for a command to finish executing before continuing. This is so if locking before sleep, the command has completed before going to sleep. Because waylock doesn't daemonize it forces systemd to wait to sleep until it hits the timeout set by [InhibitMaxDelaySec](https://www.freedesktop.org/software/systemd/man/logind.conf.html#InhibitDelayMaxSec=).

It seems like a simple enough feature to be compatible with waylock's minimalism. I would contribute it but am not familiar enough with zig. Here is swaylock's [implementation](https://github.com/swaywm/swaylock/blob/84598b08f21ae607810044f39caf9e0569cf2087/main.c#L1273).

**Comment by ifreund** [comment]
As mentioned in the `waylock(1)` man page, `setsid(1)` may be used for this purpose.

**Comment by jordanisaacs** [comment]
The issue I see, I may be wrong, is that setsid will occur immediately. While daemonizing it within waylock will let us do it *after* the lock is initialized. Thus, eg when using with with swayidle, it prevents the suspend until the lock occurs.

**Comment by jordanisaacs** [comment]
I doubt it is idiomatic zig but here is my commit, see how it daemonizes after the lock occurs: https://github.com/jordanisaacs/waylock/commit/16cf2fc344a59ad2f4fe28fc4aca80df05285a54

**Comment by ifreund** [comment]
Ah, yes good point about the potential race there. Have you hit this race in practice? Regardless this seems like an option worth adding.

**Comment by jordanisaacs** [comment]
I have not experienced it myself (haven't tried your suggested setsid solution). But this is how swaylock does it (which it seems took it from i3lock). My link in the original post above links to their code. So I would not be surprised if it has been an issue.

**Comment by jordanisaacs** [comment]
Let me know if that commit is salvageable into more idiomatic zig or if you just want to do it from scratch. If so, I can create a PR from that commit and clean it up/push it over the line.

**Comment by ifreund** [comment]
Oh, and I'll tag a new release after #48 has been implemented as well :)

**Comment by jordanisaacs** [comment]
No problem, thanks for the quick implementation! That looks much cleaner than my hacky port of swaylock's style. I'll switch to it and will post here if I encounter any issues.

Also looking forward to a new release tag as since I am maintaining waylock on nixpkgs I will be able to update it for everyone right when it releases :)
